jms/src/main/scala/akka/stream/alpakka/jms/Jms.scala

```diff
@@ -15,6 +17,19 @@ sealed trait Destination
 final case class Topic(name: String) extends Destination
 final case class Queue(name: String) extends Destination
 
+final case class JmsTextMessage(body: String, properties: Map[String, Any] = Map.empty) {
+  def withProperty(name: String, value: Any) = copy(properties = properties + (name -> value))
```

Comment:
Will do! I used the `width` since I saw if being used already in other connectors like [on the MQTT connector](https://github.com/akka/alpakka/blob/master/mqtt/src/main/scala/akka/stream/alpakka/mqtt/Mqtt.scala#L83):
```scala
final case class MqttConnectionSettings(
    broker: String,
    clientId: String,
    persistence: MqttClientPersistence,
    auth: Option[(String, String)] = None,
    socketFactory: Option[SSLSocketFactory] = None,
    cleanSession: Boolean = true,
    will: Option[Will] = None
) {
  def withBroker(broker: String) =
    copy(broker = broker)

  def withAuth(username: String, password: String) =
    copy(auth = Some((username, password)))

  def withCleanSession(cleanSession: Boolean) =
    copy(cleanSession = cleanSession)

  def withWill(will: Will) =
    copy(will = Some(will))

  def withClientId(clientId: String) =
    copy(clientId = clientId)
}
```